diff -up plasma-systemmonitor-5.21.3/src/page/PageDataObject.cpp.1~ plasma-systemmonitor-5.21.3/src/page/PageDataObject.cpp
--- plasma-systemmonitor-5.21.3/src/page/PageDataObject.cpp.1~	2021-03-16 13:38:13.000000000 +0100
+++ plasma-systemmonitor-5.21.3/src/page/PageDataObject.cpp	2021-03-29 21:49:06.933299345 +0200
@@ -31,6 +31,12 @@ QVariant converted(const QVariant &varia
         return QVariant{};
     }
 
+    // QVariant is also a bit overeager to convert non-numeric strings
+    // to the number 0
+    if (result.toString() == "0" && variant.toString() != "0") {
+        return QVariant{};
+    }
+
     if (type == QMetaType::Bool) {
         if (result.toBool()) {
             // Unfortunately, QVariant regards any string that is not "no" or "false" as true.
